src/routes/webhooks/paddle.ts

```diff
@@ -103,54 +104,75 @@ export const updateUserSubscription = async ({
     return false;
   }
 
-  const user = await con.getRepository(User).findOneBy({ id: userId });
-  if (!user) {
-    logger.error(
-      { provider: SubscriptionProvider.Paddle, data: event },
-      'User not found',
-    );
-    return false;
-  }
-
-  if (user.subscriptionFlags?.provider === SubscriptionProvider.AppleStoreKit) {
-    logger.error(
-      {
-        user,
-        data: event,
+  if (userId === 'anonymous') {
```

Comment:
What does the ID have to be `anonymous`? It can either be empty or if it does, we can just search the DB which will return not found. Normally not a fan of strongly typed strings that are used across systems.

Reply:
Yeah, basically if there is no `customData.user_id` we treat as anonymous purchase and we should create claimable item row.

Reply:
Let's create a helper in `src/common/paddle/index.ts` for handling anonymous purchase and move the logic there, it will be a bit more readable.